stan/math/opencl/kernel_generator/transpose.hpp

```diff
@@ -0,0 +1,131 @@
+#ifndef STAN_MATH_OPENCL_KERNEL_GENERATOR_TRANSPOSE_HPP
+#define STAN_MATH_OPENCL_KERNEL_GENERATOR_TRANSPOSE_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl_view.hpp>
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/opencl/kernel_generator/type_str.hpp>
+#include <stan/math/opencl/kernel_generator/name_generator.hpp>
+#include <stan/math/opencl/kernel_generator/operation_cl.hpp>
+#include <stan/math/opencl/kernel_generator/as_operation_cl.hpp>
+#include <stan/math/opencl/kernel_generator/is_valid_expression.hpp>
+#include <algorithm>
+#include <string>
+#include <set>
+#include <type_traits>
+#include <utility>
+
+namespace stan {
+namespace math {
+
+/**
+ * Represents a transpose in kernel generator expressions.
+ *
+ * Warning: transposing this expression is not supported!
+ * @tparam Derived derived type
+ * @tparam T_a type of first argument
+ * @tparam T_b type of second argument
```

Comment:
These don't match up

Reply:
Also this may be a better place for clearer names like calling `T` `Expr`

Reply:
fixed